cms/djangoapps/contentstore/management/commands/tests/test_migrate_transcripts.py

```diff
@@ -0,0 +1,212 @@
+        self.assertEqual(len(languages), 0)
+
+        # now call migrate_transcripts command and check the transcript availability
+        call_command('migrate_transcripts', unicode(self.course.id))
+
+        # check that transcripts still do not exist
+        languages = api.get_available_transcript_languages(self.video_descriptor.edx_video_id)
+        self.assertEqual(len(languages), 0)
+
+    def test_migrate_transcripts_availability(self):
+        """
+        Test migrating transcripts
+        """
+        translations = self.video_descriptor.available_translations(self.video_descriptor.get_transcripts_info())
+        self.assertItemsEqual(translations, ['hr', 'ge'])
+        self.assertFalse(api.is_transcript_available(self.video_descriptor.edx_video_id, 'hr'))
+        self.assertFalse(api.is_transcript_available(self.video_descriptor.edx_video_id, 'ge'))
+
+        # now call migrate_transcripts command and check the transcript availability
+        call_command('migrate_transcripts', unicode(self.course.id), '--commit')
+
+        self.assertTrue(api.is_transcript_available(self.video_descriptor.edx_video_id, 'hr'))
+        self.assertTrue(api.is_transcript_available(self.video_descriptor.edx_video_id, 'ge'))
+
+    def test_migrate_transcripts_idempotency(self):
+        """
+        Test migrating transcripts multiple times
+        """
+        translations = self.video_descriptor.available_translations(self.video_descriptor.get_transcripts_info())
+        self.assertItemsEqual(translations, ['hr', 'ge'])
+        self.assertFalse(api.is_transcript_available(self.video_descriptor.edx_video_id, 'hr'))
+        self.assertFalse(api.is_transcript_available(self.video_descriptor.edx_video_id, 'ge'))
+
+        # now call migrate_transcripts command and check the transcript availability
+        call_command('migrate_transcripts', unicode(self.course.id), '--commit')
+
+        self.assertTrue(api.is_transcript_available(self.video_descriptor.edx_video_id, 'hr'))
+        self.assertTrue(api.is_transcript_available(self.video_descriptor.edx_video_id, 'ge'))
+
+        # now call migrate_transcripts command again and check the transcript availability
+        call_command('migrate_transcripts', unicode(self.course.id), '--commit')
+
+        self.assertTrue(api.is_transcript_available(self.video_descriptor.edx_video_id, 'hr'))
+        self.assertTrue(api.is_transcript_available(self.video_descriptor.edx_video_id, 'ge'))
+
+        # now call migrate_transcripts command with --force-update and check the transcript availability
+        call_command('migrate_transcripts', unicode(self.course.id), '--force-update', '--commit')
+
+        self.assertTrue(api.is_transcript_available(self.video_descriptor.edx_video_id, 'hr'))
+        self.assertTrue(api.is_transcript_available(self.video_descriptor.edx_video_id, 'ge'))
+
+    def test_migrate_transcripts_logging(self):
```

Comment:
We haven't checked for any exception logs. We need to check them as well in this test or a separate test

Reply:
Done